clib/dpd.js

```diff
@@ -2,9 +2,31 @@
 
   if (!window._dpd) window._dpd = {};
 
-  var root = window.location.protocol + '//' + window.location.hostname;
-  if (window.location.port !== '') {
-    root += ':' + window.location.port;
+  var root = dpdScriptElement().getAttribute('src')
+      .replace(/(.*)(\/.*\.js)$/, '$1')
+      .replace(/^[^\/]*.js$/, '');
+
```

Comment:
Is it possible that root still not yet set in here? 

No currentScript, no id=dpd.js and script src is changed?

if yes, let's add a checking and fail back to the old code?